lib/vdc/DCCF.cpp

```diff
@@ -68,13 +68,17 @@ int DCCF::initialize(const vector<string> &paths, const std::vector<string> &opt
 
 int DCCF::initialize(const vector<string> &paths, const std::vector<string> &options, NetCDFCFCollection *ncdfc)
 {
-    if (_ncdfc) delete _ncdfc;
+    if (_ncdfc) {
+      delete _ncdfc;
+      _ncdfc = nullptr;
+    }
     _paths = paths;
     
     // Initialize the NetCDFCFCollection class.
     //
     int rc = ncdfc->Initialize(paths);
     if (rc < 0) {
+        delete ncdfc;
```

Comment:
The memory leak you are fixing here is due to the function being improperly called. This will cause a segfault if it is used correctly.

Reply:
This is a downside of using raw pointers. Do you suggest to revert my change here, or to spend some effort to clean up related code in a proper manner?

Reply:
I would at least revert the change as a small memory leak is better than a segfault, however, I believe this function is currently only used in one location so it should be very easy to fix properly.

Reply:
Good idea. Do you mind reviewing again?

Reply:
Looks good, thanks